TestHelper.psm1

```diff
@@ -73,3 +73,61 @@ function New-Nuspec
     New-Item -Path $nuspecPath -ItemType File -Force > $null
     Set-Content -Path $nuspecPath -Value $content
 }
+
+<#
+    .SYNOPSIS
+        Will attempt to download the resource designer code via PowerShellGet or Nuget package.
+    
+    .EXAMPLE
+        Get-ResourceDesigner
+
+#>
+function Get-ResourceDesigner {
+    [CmdletBinding()]
+       
+    $DesignerModuleName = 'xDscResourceDesigner'
+    $DesignerModulePath = "$env:USERPROFILE\Documents\WindowsPowerShell\Modules\$DesignerModuleName"
+    $OutputDirectory = "$(Split-Path -Path $DesignerModulePath -Parent)\"
+
+    if ($env:APPVEYOR) {
+        if (Test-Path -Path $DesignerModulePath)
+        {
+            # Remove any installed version of the DscResourceDesigner module
+            Remove-Item -Path $DesignerModulePath -Recurse -Force
+        }
+  
+        # Is PowerShellGet module installed?
+        if (@(Get-Module -Name PowerShellGet -ListAvailable).Count -ne 0)
+        {
+            Import-Module PackageManagement
+
+            # Make sure the Nuget Package provider is initialized.
+            Get-PackageProvider -name nuget -ForceBootStrap -Force
+
+            # PowerShellGet is available - use that
+            Import-Module PowerShellGet
+
+            # Install the module - make sure we terminate if it fails.
+            Install-Module -Name $DesignerModuleName -Force
+        }
+        else
+        {
+            # PowerShellGet module isn't available, so use Nuget directly to download it
+            $nugetSource = 'https://www.powershellgallery.com/api/v2'
+            $nugetPath = 'nuget.exe'
+            & "$nugetPath" @('install',$DesignerModuleName,'-source',$nugetSource,'-outputDirectory',$OutputDirectory,'-ExcludeVersion')
+            $ExitCode = $LASTEXITCODE
+
+            if ($ExitCode -ne 0)
+            {
+                throw (
+                    'Module installation using Nuget of {0} failed with exit code {1}.' `
+                        -f $DesignerModuleName,$ExitCode
+                    )
+            }
+        }
+    }
+
+    # Import the module using the name
+    Import-Module -Name $DesignerModuleName -Force
```

Comment:
I'm not convinced this is the right approach. Think about it this way:

A new contributor wants to contribute to xSharePoint (or any other resource), clones the repo, runs tests and half of them fail. There's no information why they fail other than "Test-xDscResource" is not recognized. They likely are not aware of what this cmdlet does and that it belongs to the xDscResourceDesigner...

I'd find this experience very discouraging. 

At the very least, we should detect whether Import-Module fails and if it does (and we aren't on appveyor), display warning that we may need to install xDscResourceDesigner manually.

Having said that, I'd actually say that there's nothing wrong in downloading xDscResourceDeisgner on non-appveyor machines. After all, our tests are modifying states of users' machines and they should be aware of that when running them (or any other script for that matter).

That'd be my preference, but I'd like to hear some other opinions on this before proceeding. @TravisEz13 ?